c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -299,24 +299,28 @@ impl<'tcx> TypeChecker<'tcx, '_> {
 
                     Some(Callee::Trivial) => {}
 
-                    Some(Callee::Malloc) => {
-                        let destination = self.acx.c_void_casts.get_or_default_to(&destination);
-                        self.visit_place(destination, Mutability::Mut);
-                    }
-
-                    Some(Callee::Calloc) => {
-                        let destination = self.acx.c_void_casts.get_or_default_to(&destination);
-                        self.visit_place(destination, Mutability::Mut);
+                    Some(Callee::Malloc | Callee::Calloc) => {
+                        let out_ptr = self
+                            .acx
+                            .c_void_casts
+                            .get_adjusted_place(CVoidCastDirection::From, destination);
+                        self.visit_place(out_ptr, Mutability::Mut);
                     }
                     Some(Callee::Realloc) => {
-                        let destination = self.acx.c_void_casts.get_or_default_to(&destination);
-                        let input_ptr_pl = args[0].place().unwrap();
-                        let first_arg = self.acx.c_void_casts.get_or_default_to(&input_ptr_pl);
-                        self.visit_place(destination, Mutability::Mut);
-                        let pl_lty = self.acx.type_of(destination);
+                        let out_ptr = self
+                            .acx
+                            .c_void_casts
+                            .get_adjusted_place(CVoidCastDirection::From, destination);
```

Comment:
this new naming `get_adjusted_place` doesn't make it clear that it will default to the input `place` if there's been more adjustment. consider renaming to make this clear so that one doesn't have to look up the definition

Reply:
What I meant by `adjusted` is that the post-`*c_void` fixes is the adjusted `Place`, whether or not it has to be adjusted.  It's in the documentation for that function.  If you have a name that better indicates that, I can change it, but I'm not sure what to name it to otherwise.

Reply:
i think appending `_or_default_to` or something similar would make it much clearer

Reply:
I can switch that if you think it's a lot clearer, though I thought it was clear enough already.  The adjustment can be zero/nothing.

Reply:
i do think it's a lot clearer

Reply:
Fixed in eb0840379814d745288f4ac165788c67b4c1ef64.